File: w09/d01/joshua_oynick/mail_items/lib/mailbox.rb
class Mailbox

  attr_accessor :mail_items, :accept_mail, :packages, :postcards

  def initialize
    @mail_items = []
    @num_packages = 0
    @num_postcards = 0

  end

  def accept_mail(mail_item)
    if mail_item.is_a?(Postcard)
      @mail_items.push(mail_item)
      @num_postcards += 1
    elsif mail_item.is_a?(Package)
      @mail_items.push(mail_item)
      @num_packages += 1
    end

  end

  def open_mail
    @mail_items.shift
  end

  def to_s
    "The mailbox has #{@num_postcards} postcards and #{@num_packages} packages."
  end

end

File: w09/d03/jimmy_davis/mail_objects_app/app/assets/javascripts/mailbox.js
function Mailbox() {
  this.mailItems = [];
}

Mailbox.prototype.acceptMail = function(item) {
  if ( item.constructor == Postcard || item.constructor == Package ) {
    this.mailItems.push(item);
  }
  else {
    throw "Must be a mailitem (either a package or postcard)";
  }
};

Mailbox.prototype.openMail = function() {
  this.mailItems.shift();
};

Mailbox.prototype.toString = function() {
  var postcards = 0;
  var packages = 0;
  for( i = 0; i < this.mailItems.length; i++ ) {
    if( this.mailItems[i].constructor == Postcard ) {
      postcards++;
    }
    else {
      packages++;
    }
  }
  return "The mailbox has " + postcards + " postcards and " + packages + " packages";
};
